### DISCUSSION TITLE
Create a way to generate more consistent payload strings

### DISCUSSION THREAD

**Discussion by hawk4031** [Code-and-Response/ClusterDuck-Protocol#430] [Ideas]
I'd like to put in a proposal to build in a way for the core CDP to have a way to generate a consistent payload format. Currently, our examples contain payload strings that are generated in the example (a great place to look is at the T-Beam GPS example). My proposal here is to introduce a new class (or a function in an existing class) that users can use in their code to create a consistent payload string with key/value pairs. 

This will provide benefit in two areas: for any event types/topics we support, we can leverage this method to generate payload strings that follow a defined format that we can decide on, and for anyone creating their own payload information, they can also guarantee a standardized format in their output. 

An example:

For GPS data, we can have a core CDP method that generates a standard GPS payload that contains Latitude, Longitude, and Altitude (much like the T-Beam GPS example does already). For each key/value in the payload you want, you can call a function that you pass your key name and the value into and it will spit out a string in the correct format. We could even go so far as to pass an array of keys and values into a method and have it generate the entire payload string. 

Happy to discuss more and provide some examples.

**Comment by BrentonPoke** [comment]
Isn't payload up to the developer? I used a JSON string as my payload. Others may want to use MessagePack, or something completely different as briefly discussed in #105. But others may want protocol buffers, BSON or CBOR. [This one lib will do it all](https://github.com/nlohmann/json), but I just used ArduinoJSON.
